nemo/collections/common/tokenizers/text_to_speech/tts_tokenizers.py

```diff
@@ -561,8 +551,11 @@ def __init__(
 
         if punct:
             if non_default_punct_list is not None:
-                self.PUNCT_LIST = non_default_punct_list
-            tokens.update(self.PUNCT_LIST)
+                self.punct_list = non_default_punct_list
+            else:
+                self.punct_list = get_ipa_punctuation_list(locale)
```

Comment:
Do we have a check that the locale given is valid? We should make sure to catch if a user has typoed instead of silently returning a default punct list.

Reply:
Added